src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingsService.java

```diff
@@ -127,17 +132,20 @@ private void setFilingApiData(
 
     if (submissionDto.isForUpdate()) {
       getPublicAndPrivateData(submissionDto.getEntityNumber(), passThroughTokenHeader);
-      //Call a method to populate UpdateSubmission
-      filing.setKind(FILING_KIND_OVERSEAS_ENTITY_UPDATE);
       var publicPrivateDataCombiner = new PublicPrivateDataCombiner(
-          publicDataRetrievalService, privateDataRetrievalService, salt);
+              publicDataRetrievalService, privateDataRetrievalService, salt);
+
+      filing.setKind(FILING_KIND_OVERSEAS_ENTITY_UPDATE);
+
+      var updateSubmission = new UpdateSubmission();
+
```

Comment:
set the updateSubmission.setEntityNumber(overseasEntitySubmisisonDto.getEntityNumber());